### DISCUSSION TITLE
Unable to create a .icc file

### DISCUSSION THREAD

**Discussion by saketsneha** [ImageMagick/ImageMagick#2311] [Help]
I want to implement the same stuff as mentioned here https://stackoverflow.com/questions/6740441/how-to-set-a-color-profile-with-exiftool using im4java .I have tried multiple ways to do that .This is not a java code but a kotlin code .One of them is-

The command that I want to run - **exiftool -icc_profile -b -w icc somefile.jpg**
```
fun readICCImage(imageFile: File, externalId: String) {
            val output: List<String>
            val outputConsumer = ArrayListOutputConsumer()
            val exiftoolcmd = ExiftoolCmd()
            exiftoolcmd.setOutputConsumer(outputConsumer)
            val operation = ETOperation()
            operation.addRawArgs("-icc_profile")
            operation.addRawArgs("-b")
            operation.addRawArgs("-w icc")
            operation.addImage(imageFile.absolutePath)
            try {
                exiftoolcmd.createScript("myscript.sh", operation)
                exiftoolcmd.run(operation)
            } catch (e: Exception) {
                log.error { "Could 1not run exif command=$imageFile externalId=$externalId" }
                throw RuntimeException(e)
            }
            output = outputConsumer.output
            println(output)
        }
```

The myscript.sh shell script that gets created is correct and when I run that script it works.But while running the function , it does not create the output file , rather just outputs the content that should go into the file Can someone suggest something ?

**Comment by dlemstra** [comment]
We do not maintain the `exiftool` program so I am wondering why you are asking the question here?

**Comment by snibgo** [comment]
@saketsneha: The question seems to be about exiftool or java or kotlin, with no relevance to ImageMagick.
